### PR TITLE
test: e2e golden testing for csv command

### PR DESCRIPTION
**Pull request by Bobgy**
Set up golden file testing for `go-licenses csv` command on example modules.
Update golden files by running: `go test --update .`

The example modules are copied from
* https://github.com/google/go-licenses/pull/82/files#diff-94148d1a2f964b0818fb803205c7eeb784f34bdb714da3d3feda7706d18cfd17
* https://github.com/google/go-licenses/pull/98

4 added examples:
* hello01 - hello world module with no dependencies
* cli02 - module that have dependencies similar to a CLI tool generated by cobra
* vendored03 - module that vendors its dependencies
* replace04 - module using replace directive

Generated licenses.csv golden files will be good examples to demonstrate new features implemented in the v2 proposal: #70 and ensure that there are no regressions.

### CODE REVIEW

**Comment by Bobgy** [comment]
@wlynch for review

File: e2e_test.go [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+		{workdir: "testdata/modules/cli02"},
+		{workdir: "testdata/modules/vendored03"},
+		{workdir: "testdata/modules/replace04"},
+	}
+	originalWorkDir, err := os.Getwd()
+	if err != nil {
+		t.Fatal(err)
+	}
+	// This builds and installs go-licenses CLI to $(go env GOPATH)/bin.
+	cmd := exec.Command("go", "install", ".")
```
Comment (wlynch) [inline e2e_test.go:46]: Let's `go build -o` into a tmp directory instead so we don't mutate a users copy of the tool.

File: e2e_test.go [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+
+	"github.com/google/go-cmp/cmp"
+)
+
+var update = flag.Bool("update", false, "update golden files")
+
+func TestCsvCommandE2E(t *testing.T) {
+	var tests = []struct {
+		workdir string
+	}{
```
Comment (wlynch) [inline e2e_test.go:35]:
```suggestion
	workdir := []string {
```

**Review by wlynch — COMMENTED** [review]
Mostly lgtm, just minor comments.

File: e2e_test.go (hunk 1 above)
Comment (Bobgy) [reply e2e_test.go]: Good point! Fixed

File: e2e_test.go (hunk 2 above)
Comment (Bobgy) [reply e2e_test.go]: Applied the suggestion : )

**Review by wlynch — APPROVED** [review]

**Pull request merged**